opentelemetry-api/src/baggage.rs

```diff
@@ -494,4 +522,50 @@ mod tests {
         let baggage = data.into_iter().collect::<Baggage>();
         assert_eq!(baggage.len(), 3)
     }
+
+    #[test]
+    fn serialize_baggage_as_string() {
+        // Empty baggage
+        let b = Baggage::default();
+        assert_eq!("", b.to_string());
+
+        // "single member empty value no properties"
+        let mut b = Baggage::default();
+        b.insert("foo", StringValue::from(""));
+        assert_eq!("foo=", b.to_string());
+
+        // "single member no properties"
+        let mut b = Baggage::default();
+        b.insert("foo", StringValue::from("1"));
+        assert_eq!("foo=1", b.to_string());
+
+        // "URL encoded value"
+        let mut b = Baggage::default();
+        b.insert("foo", StringValue::from("1=1"));
+        assert_eq!("foo=1%3D1", b.to_string());
+
+        // "single member empty value with properties"
+        let mut b = Baggage::default();
+        b.insert_with_metadata(
+            "foo",
+            StringValue::from(""),
+            BaggageMetadata::from("red;state=on"),
+        );
+        assert_eq!("foo=;red;state=on", b.to_string());
+
+        // "single member with properties"
+        let mut b = Baggage::default();
+        b.insert_with_metadata("foo", StringValue::from("1"), "red;state=on;z=z=z");
+        assert_eq!("foo=1;red;state=on;z=z=z", b.to_string());
+
+        // "two members with properties"
+        let mut b = Baggage::default();
+        b.insert_with_metadata("foo", StringValue::from("1"), "red;state=on");
+        b.insert_with_metadata("bar", StringValue::from("2"), "yellow");
+        let stringified = b.to_string();
+        assert!(
+            stringified == "bar=2;yellow,foo=1;red;state=on"
+                || stringified == "foo=1;red;state=on,bar=2;yellow"
+        )
```

Comment:
Removing the sort makes `.to_string()` non deterministic, which is all fine and dandy but testing is a bit harder, idk, this feels a bit heavy handed but I'll let you be the judge of that 😂

Reply:
I'd maybe do something like `stringified.contains("bar=2;yellow") && stringified.contains("foo=1;red;state=on")` which feels more robust/obvious. But I do think shifting the complexity from library to test is probably for the best!